ext/src/ruby_api/engine.rs

```diff
@@ -16,7 +16,7 @@ lazy_static::lazy_static! {
 /// @yard
 /// Represents a Wasmtime execution engine.
 /// @see https://docs.rs/wasmtime/latest/wasmtime/struct.Engine.html Wasmtime's Rust doc
-#[magnus::wrap(class = "Wasmtime::Engine", free_immediately)]
+#[magnus::wrap(class = "Wasmtime::Engine", free_immediately, frozen_shareable)]
```

Comment:
The `timer_task` field uses a `RefCell` which I think isn't thread safe. IIUC, by making this `frozen_shareable`, we're saying that instances are frozen and can be used across threads _at the same time_.

Should we change that `RefCell` for a `Mutex`?

Reply:
Damn! Would be nice if Magnus required `Sync` on the type as soon as we use this flag.

Seems it does, but only on the `DataTypeFunctions`: https://github.com/matsadler/magnus/blob/f7aa98abbed4a1d6e2b3aed19c9e1f2064f0350e/src/typed_data.rs#L364-L375.